clang-tools-extra/clang-tidy/daedalean/TypeConversionsCheck.cpp

```diff
@@ -207,7 +207,9 @@ void TypeConversionsCheck::handleImplicitCast(clang::ASTContext *context,
                                               QualType sourceType,
                                               QualType destType,
                                               SourceLocation location) {
-  if (destType.getCanonicalType() == sourceType.getCanonicalType()) {
+  sourceType = sourceType.getCanonicalType();
+  destType = destType.getCanonicalType();
+  if (destType == sourceType) {
```

Comment:
I wonder why they have this  distinction between canonical and non-canonical types...

Reply:
For example some checkers not complaining on template arguments or apply different rules

Reply:
In my opinion that is a characteristic of the ast node, not part of the type of the node. Although I guess I just don't have enough context about this.